lib/app/app.go

```diff
@@ -0,0 +1,184 @@
+	flag.BoolVar(&Flags.AmRoot, "amroot", false, "am I root node")
+	flag.BoolVar(&Flags.TestConnect, "test_connect", false, "test connecting and disconnecting")
+	flag.StringVar(&Flags.Mode, "mode", Flags.Mode, "Run the app in [server,client] mode")
+	flag.StringVar(&Flags.Name, "name", Flags.Name, "Name of the node")
+	flag.StringVar(&Flags.Server, "server", "", "the timestamping servers to contact")
+}
+
+/*
+ * Reads in the config from deterlab and for the application -
+ * also parses the init-flags
+ * It first reads the configuration of deterlab, in case the
+ * application needs any of these configuration-options, then
+ * loads over that the configuration of the app.toml
+ */
+func ReadConfig(conf interface{}, dir ...string) {
+	var err error
+	err = ReadTomlConfig(conf, "deter.toml", dir...)
+	if err != nil {
+		dbg.Lvl2("Couldn't load deter-config")
+	}
+	err = ReadTomlConfig(conf, "app.toml", dir...)
+	if err != nil {
+		log.Fatal("Couldn't load app-config-file in exec")
+	}
+	debug := reflect.ValueOf(conf).Elem().FieldByName("Debug")
+	if debug.IsValid(){
+		dbg.DebugVisible = debug.Interface().(int)
+	}
+
+	flag.Parse()
+
+	dbg.Lvl3("Running", Flags.Hostname, "with logger at", Flags.Logger)
+	ConnectLogservers()
+	ServeMemoryStats()
+}
+
+/*
+ * Connects to the logservers for external logging
+ */
+func ConnectLogservers() {
+	// connect with the logging server
+	if Flags.Logger != "" && Flags.AmRoot{
+		// blocks until we can connect to the flags.Logger
+		dbg.Lvl3(Flags.Hostname, "Connecting to Logger", Flags.Logger)
+		lh, err := logutils.NewLoggerHook(Flags.Logger, Flags.Hostname, "unknown")
+		if err != nil {
+			log.WithFields(log.Fields{
+				"file": logutils.File(),
+			}).Fatalln("Error setting up logging server:", err)
+		}
+		log.AddHook(lh)
+		//log.SetOutput(ioutil.Discard)
+		//fmt.Println("exiting flags.Logger block")
+		dbg.Lvl4(Flags.Hostname, "Done setting up hook")
+	} else {
+		dbg.Lvl4("Not connecting to logger - logger:", Flags.Logger, "AmRoot:", Flags.AmRoot)
+	}
+}
+
+/*
+ * Opens a port at 'flags.Hostname + 1' and serves memory-statistics of this process
+ */
+func ServeMemoryStats() {
+	if Flags.Mode == "server" {
+		if Flags.PhysAddr == "" {
+			h, _, err := net.SplitHostPort(Flags.Hostname)
+			if err != nil {
+				log.Fatal(Flags.Hostname, "improperly formatted hostname", os.Args)
+			}
+			Flags.PhysAddr = h
+		}
+
+		// run an http server to serve the cpu and memory profiles
+		go func() {
+			_, port, err := net.SplitHostPort(Flags.Hostname)
+			if err != nil {
+				log.Fatal(Flags.Hostname, "improperly formatted hostname: should be host:port")
+			}
+			p, _ := strconv.Atoi(port)
+			// uncomment if more fine grained memory debuggin is needed
+			//runtime.MemProfileRate = 1
+			res := http.ListenAndServe(net.JoinHostPort(Flags.PhysAddr, strconv.Itoa(p + 2)), nil)
+			dbg.Lvl3("Memory-stats server:", res)
+		}()
+	}
+}
```

Comment:
```ConnectLogservers``` is only used for the graph and the monitoring-function. Once your ```monitor``` is merged, we can talk about how to take that out.

```ServeMemoryStats``` could probably be taken away - I'd like to keep it for the moment, just in case we need it for further statistics.